core/src/Pos/Core/Block/Union/Types.hs

```diff
@@ -1,14 +1,17 @@
--- the Getter instances
-{-# OPTIONS_GHC -Wno-redundant-constraints #-}
+{-# OPTIONS_GHC -Wno-redundant-constraints #-} -- for the Getter instances
+
+{-# LANGUAGE TypeOperators #-}
 
 -- | Union of blockchain types.
 
 module Pos.Core.Block.Union.Types
```

Comment:
Unfortunately, I ended up squashing everything from `Pos.Core.Block.Union.Instances` and `Pos.Core.Block.Union.Lens` into here so it gets a little messy. It might be worth coming back to this later and refactoring. i.e. splitting up `Types` into different modules such as `BlockHeader`, `GenesisBlockchain`, etc.

Reply:
Yeah this is a bit crowded. Since we'll be pulling these modules out into the new `chain` package relatively soon, it seems to me like we can leave stuff here for now, and split things up properly when we migrate them to `chain`. Thoughts, @erikd ?

Reply:
Yep, keep them like this for now.